uvtools/dspec.py

```diff
@@ -189,21 +275,87 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
         a2[uthresh:lthresh] = 0
         area = np.outer(a1, a2)
 
+        uthresh_fg, lthresh_fg = calc_width(fg_restore_size[0], real_delta[0], data.shape[0])
+        a1fg = np.ones(data.shape[0], dtype=np.int)
+        a1fg[uthresh_fg:lthresh_fg] = 0
+        uthresh_fg, lthresh_fg = calc_width(fg_restore_size[1], real_delta[1], data.shape[1])
+        a2fg = np.ones(data.shape[-1], dtype=np.int)
+        a2fg[uthresh_fg:lthresh_fg] = 0
+        area_fg = np.outer(a1fg, a2fg)
+
+        fc = []
```

Comment:
please add a comment here that states that this block is only needed for `linear filter`

Reply:
Done.